parrot/src/pfs_table.cc

```diff
@@ -1599,8 +1647,11 @@ int pfs_table::mkalloc( const char *n, pfs_ssize_t size, mode_t mode )
 {
 	pfs_name pname;
 	int result=-1;
+	char dirname[PFS_PATH_MAX];
+
+	path_dirname(n,dirname);
 
-	if(resolve_name(1,n,&pname)) {
+	if(resolve_name(0,dirname,&pname,X_OK|W_OK) && resolve_name(0,n,&pname,F_OK)) { //TODO check this
```

Comment:
This seems to be a common check in the code: modifying/creating the directory entry for the given path. I suggest creating a new flag, `E_OK`, which tests this and combine the check in a single call to `resolve_name`.